refiner-lib/src/refiner_inner.rs

```diff
@@ -427,6 +432,38 @@ fn build_transaction(
                         txs.get(&hash),
                     )?
                 }
+                InnerTransactionKind::Deploy | InnerTransactionKind::DeployErc20 => {
+                    hash = virtual_receipt_id.0.try_into().unwrap();
+                    tx = tx.hash(hash).from(near_account_to_evm_address(
+                        outcome.receipt.predecessor_id.as_bytes(),
+                    ));
+
+                    tx = tx
+                        .to(None)
+                        .nonce(U256::zero())
+                        .gas_limit(U256::from(u64::MAX))
+                        .max_priority_fee_per_gas(U256::zero())
+                        .max_fee_per_gas(U256::zero())
+                        .value(Wei::zero())
+                        .input(vec![])
+                        .access_list(vec![])
+                        .tx_type(0xff)
```

Comment:
Found this paragraph at the end of the article:

'Backwards Compatibility

Clients can differentiate between the legacy transactions and typed transactions by looking at the first byte. If it starts with a value in the range [0, 0x7f] then it is a new transaction type, if it starts with a value in the range [0xc0, 0xfe] then it is a legacy transaction type. 0xff is not realistic for an RLP encoded transaction, so it is reserved for future use as an extension sentinel value.'

Reply:
So far what I have found is that:
1. TransactionType is still not an enforced standard.
2. As mentioned before, [0, 0x7f] is a new transaction type and [0xc0, 0xfe] is a legacy transaction type
3. 'Txn Type: 2' (0x02) is the most common type that I have found.

If we need to use a default here my guess is that the end of the range of the new type will be the best to follow our own previous behave. I.e 0x7f

There are existing parts of the code (no new code) in this file that are also using the 0xff. We should probably change them all if we are going to proceed.

Reply:
I looked into this more and came to the conclusion that we should leave the `0xff` value. See below for my reasoning.

Even though the EIP-2718 spec says that 0 is allowed as a type id, it looks like all the major clients have taken the convention that 0 is a sentinel value for legacy transactions used in unified models of Ethereum transactions (i.e. the 0 means "there is no type envelope" as opposed to "the type id is 0").

So we could use 0 as the default value and I expect all consumers of the data would handle that fine. However they may be concerned by the signature values being all zeros too because they will assume this is a legacy transaction, which is supposed to be signed. Of course the reason there is no signature is because the real transaction was on NEAR and signed using a totally different signature scheme that we cannot map onto these pseudo-Ethereum transactions.

Due to the missing signature, it makes sense we might want to pick a different type id so that clients do not expect the transaction to be signed. However, then we are back to the problem that introducing a new type id could conflict with future Ethereum usages, and clients might need to modify their code to not panic when they see a type id other than 0, 1, or 2.

Ultimately, aurorascan has already started using data output from the refiner and as far as I know is almost done. This means they must have found a way to make the `0xff` value work for them. Therefore, I think the safest option is to just leave it the way it is since there is no "good" choice.

I'll mark this comment as resolved and we'll leave the `0xff` until someone tells us otherwise.